docs/guides/middleware.md

```diff
@@ -38,28 +53,44 @@ Mediators are called on the `Player` first, run through middleware (from left to
+### Middleware Getters
 
-```
-               mediate to tech
-               +------------->
+Getters are called on the `Tech` first and are run though middleware in reverse of the order they were registered in (from right to left in the diagram) before returning the result to the `Player`.
 
+### Middleware Mediators
+
+Mediators are methods that not only change the state of the `Tech`, but also return some value back to the `Player`. Currently, these are `play` and `pause`.
+
+Mediators are called on the `Player` first, run through middleware in the order they were registered (from left to right in the below diagram), then called on the `Tech`. The result is returned to the `Player` unchanged, while calling the middleware in the reverse order of how they were registered (from right to left in the diagram.) For more information on mediators, check out the [mediator section](#termination-and-mediators).
+
+```
 +----------+                      +----------+
 |          |                      |          |
-|          +-----call{method}----->          |
+|          +---mediate-to-tech---->          |
 |  Player  |                      |   Tech   |
-|          <-------{method}-------+          |
+|          <--mediate-to-player---+          |
 |          |                      |          |
 +----------+                      +----------+
+```
+
+### Termination and Mediators
+
+Mediators make a round trip: starting at the `Player`, mediating to the `Tech` and returning the result to the `Player` again. A `call{method}` method must be supplied by the middleware which is used when mediating to the `Tech`. On the way back to the `Player`, the `{method}` will be called instead, with 2 arguments: `terminated`, a Boolean indicating whether a middleware terminated during the mediation to the tech portion, and `value`, which is the value returned from the `Tech`.
 
-              <---------------+
-              mediate to player
+```
++----------+                      +----------+
+|          |                      |          |
+|          +----+call{method}+---->          |
+|  Player  |                      |   Tech   |
+|          <------+{method}+------+          |
+|          |                      |          |
++----------+                      +----------+
 
 ```
 
-Mediators make a round trip: starting at the `Player`, mediating to the `Tech` and returning the result to the `Player` again. A `call{method}` method must be supplied by the middleware which is used when mediating to the `Tech`. On the way back to the `Player`, the `{method}` will be called instead, with 2 arguments: `terminated`, a Boolean indicating whether a middleware terminated during the mediation to the tech portion, and `value`, which is the value returned from the `Tech`. A barebones example of a middleware with Mediator methods is:
+A skeleton of a middleware with Mediator methods is given below:
```

Comment:
I noticed that this example uses `callPlay` and `pause`. Should the `pause` be `play`?

Reply:
good catch